testsuite/tests/conftest.py

```diff
@@ -166,11 +167,15 @@ def keycloak(request, testconfig, blame, skip_or_fail):
         )
 
         info.commit()
-        return info
+        yield info
     except KeycloakAuthenticationError:
-        return skip_or_fail("Unable to login into SSO, please check the credentials provided")
+        yield skip_or_fail("Unable to login into SSO, please check the credentials provided")
     except KeyError as exc:
-        return skip_or_fail(f"SSO configuration item is missing: {exc}")
+        yield skip_or_fail(f"SSO configuration item is missing: {exc}")
+
+    # Realm cleanup
+    if info is not None:
```

Comment:
if you need this you can use addfinalizer instead
```
request.addfinalizer(info.delete)
info.commit()
return info
```

Reply:
and as a bonus you wont need to change the returns

Reply:
I was thinking about using finalizers but internet told me that `yield` is preferred if one deals with single resource. Agree that that `None` check is not nice but it has to be there since the cleanup also gets executed in case of exception (because I had to replace `return` with `yield` in that exception clauses as well). If you consider the finalizer approach nicer in this particular case I will refactor. I think you are actually right, it is nicer.

Reply:
Thinking about it, the downside would be that `info.delete()` is to be called even in case of exception on `info` object creation so there might be an (additional) exception raised on cleanup. Which is acceptable I think.

Reply:
Refactored using finalizer approach